### PR TITLE
rearrange parallelism in AKS machine pool e2e tests

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

I have been seeing some flakes in the e2e tests for AKS machine pool scaling recently and it's hard to know exactly what's going wrong because of the way Ginkgo handles how these tests are parallelized (I think).

The tests are currently structured like this:
```
for each machine pool in parallel:
  scale up
wait

for each machine pool in parallel:
  scale down
wait

for each machine pool in parallel:
  scale to zero
wait

for each machine pool in parallel:
  scale to where we started
wait
```

It seems when Ginkgo sees a failure in any of these steps, the test continues to the next step anyway, which can cause some misleading error messages and the collected artifacts don't reflect the state of the resources at the time the test failed.

This PR reformulates the tests to work like this:
```
for each machine pool in parallel:
  scale up
  scale down
  scale to zero
  scale to where we started
wait
```
This way, Ginkgo does short-circuit when any of the intermediate steps fail because all the steps are done in the same goroutine for an individual MachinePool.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate <-- for improved signal on release branches.

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/aks_machinepools.go [hunk 1]
```diff
@@ -47,109 +47,92 @@ func AKSMachinePoolSpec(ctx context.Context, inputGetter func() AKSMachinePoolSp
 	input := inputGetter()
 	var wg sync.WaitGroup
 
-	originalReplicas := map[types.NamespacedName]int32{}
```
Comment (nojnhuh) [inline test/e2e/aks_machinepools.go:50]: Most of the changes in this file are indentation, so I'd suggest ticking the "hide whitespace" checkbox.

**Comment by nojnhuh** [comment]
/cherry-pick release-1.15

Skipping 1.14 because I haven't seen the flakes there and this will probably not cherry pick cleanly and I'm lazy.

**Comment by nojnhuh** [comment]
AKS job failure looks like the same flake I've been seeing recently and these changes definitely helped narrow it down! I have a fix for that open in #4875.

In the meantime,
/retest

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**